@bogdandrutu should I run that from root, or the `receiver/statsdreceiver` directory? `make gotidy` seems to partially fail and reports a ton of changes (maybe due to the partial failure).

```
→ make gotidy 
make for-all CMD="rm -fr go.sum"
make[1]: Entering directory 'opentelemetry-collector-contrib'
running rm -fr go.sum in ./exporter/alibabacloudlogserviceexporter
running rm -fr go.sum in ./exporter/awsxrayexporter
running rm -fr go.sum in ./exporter/azuremonitorexporter
running rm -fr go.sum in ./exporter/carbonexporter
running rm -fr go.sum in ./exporter/elasticexporter
running rm -fr go.sum in ./exporter/honeycombexporter
running rm -fr go.sum in ./exporter/jaegerthrifthttpexporter
running rm -fr go.sum in ./exporter/kinesisexporter
running rm -fr go.sum in ./exporter/lightstepexporter
running rm -fr go.sum in ./exporter/newrelicexporter
running rm -fr go.sum in ./exporter/sapmexporter
running rm -fr go.sum in ./exporter/sentryexporter
running rm -fr go.sum in ./exporter/signalfxexporter
running rm -fr go.sum in ./exporter/splunkhecexporter
running rm -fr go.sum in ./exporter/stackdriverexporter
running rm -fr go.sum in ./extension/observer
running rm -fr go.sum in ./extension/observer/hostobserver
running rm -fr go.sum in ./extension/observer/k8sobserver
running rm -fr go.sum in ./internal/common
running rm -fr go.sum in ./internal/common/awsxray/testdata/sampleapp
running rm -fr go.sum in ./internal/common/awsxray/testdata/sampleserver
running rm -fr go.sum in ./processor/k8sprocessor
running rm -fr go.sum in ./processor/metricstransformprocessor
running rm -fr go.sum in ./processor/resourcedetectionprocessor
running rm -fr go.sum in ./receiver/awsecscontainermetricsreceiver
running rm -fr go.sum in ./receiver/awsxrayreceiver
running rm -fr go.sum in ./receiver/carbonreceiver
running rm -fr go.sum in ./receiver/collectdreceiver
running rm -fr go.sum in ./receiver/k8sclusterreceiver
running rm -fr go.sum in ./receiver/kubeletstatsreceiver
running rm -fr go.sum in ./receiver/prometheusexecreceiver
running rm -fr go.sum in ./receiver/receivercreator
running rm -fr go.sum in ./receiver/redisreceiver
running rm -fr go.sum in ./receiver/sapmreceiver
running rm -fr go.sum in ./receiver/signalfxreceiver
running rm -fr go.sum in ./receiver/simpleprometheusreceiver
running rm -fr go.sum in ./receiver/statsdreceiver
running rm -fr go.sum in ./receiver/wavefrontreceiver
running rm -fr go.sum in ./testbed
make[1]: Leaving directory 'opentelemetry-collector-contrib'
make for-all CMD="go mod tidy"
make[1]: Entering directory 'opentelemetry-collector-contrib'
running go mod tidy in ./exporter/alibabacloudlogserviceexporter
go: downloading golang.org/x/tools v0.0.0-20200724022722-7017fd6b1305
running go mod tidy in ./exporter/awsxrayexporter
go: downloading github.com/Azure/go-autorest/autorest/mocks v0.3.0
go: downloading github.com/googleapis/gnostic v0.4.1
go: finding module for package github.com/googleapis/gnostic/OpenAPIv2
go: downloading github.com/googleapis/gnostic v0.5.1
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter tested by
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter.test imports
	go.opentelemetry.io/collector/config/configcheck tested by
	go.opentelemetry.io/collector/config/configcheck.test imports
	go.opentelemetry.io/collector/service/defaultcomponents imports
	go.opentelemetry.io/collector/receiver/prometheusreceiver imports
	github.com/prometheus/prometheus/discovery imports
	github.com/prometheus/prometheus/discovery/kubernetes imports
	k8s.io/client-go/kubernetes imports
	k8s.io/client-go/discovery imports
	github.com/googleapis/gnostic/OpenAPIv2: module github.com/googleapis/gnostic@latest found (v0.5.1), but does not contain package github.com/googleapis/gnostic/OpenAPIv2
Makefile:59: recipe for target 'for-all' failed
make[1]: *** [for-all] Error 1
make[1]: Leaving directory 'opentelemetry-collector-contrib'
Makefile:50: recipe for target 'gotidy' failed
make: *** [gotidy] Error 2
```

`git status`:

```
→ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   exporter/alibabacloudlogserviceexporter/go.sum
	modified:   exporter/awsxrayexporter/go.sum
	deleted:    exporter/azuremonitorexporter/go.sum
	deleted:    exporter/carbonexporter/go.sum
	deleted:    exporter/elasticexporter/go.sum
	deleted:    exporter/honeycombexporter/go.sum
	deleted:    exporter/jaegerthrifthttpexporter/go.sum
	deleted:    exporter/kinesisexporter/go.sum
	deleted:    exporter/lightstepexporter/go.sum
	deleted:    exporter/newrelicexporter/go.sum
	deleted:    exporter/sapmexporter/go.sum
	deleted:    exporter/sentryexporter/go.sum
	deleted:    exporter/signalfxexporter/go.sum
	deleted:    exporter/splunkhecexporter/go.sum
	deleted:    exporter/stackdriverexporter/go.sum
	deleted:    extension/observer/go.sum
	deleted:    extension/observer/hostobserver/go.sum
	deleted:    extension/observer/k8sobserver/go.sum
	modified:   go.sum
	deleted:    internal/common/awsxray/testdata/sampleapp/go.sum
	deleted:    internal/common/awsxray/testdata/sampleserver/go.sum
	deleted:    internal/common/go.sum
	deleted:    processor/k8sprocessor/go.sum
	deleted:    processor/metricstransformprocessor/go.sum
	deleted:    processor/resourcedetectionprocessor/go.sum
	deleted:    receiver/awsecscontainermetricsreceiver/go.sum
	deleted:    receiver/awsxrayreceiver/go.sum
	deleted:    receiver/carbonreceiver/go.sum
	deleted:    receiver/collectdreceiver/go.sum
	deleted:    receiver/k8sclusterreceiver/go.sum
	deleted:    receiver/kubeletstatsreceiver/go.sum
	deleted:    receiver/prometheusexecreceiver/go.sum
	deleted:    receiver/receivercreator/go.sum
	deleted:    receiver/redisreceiver/go.sum
	deleted:    receiver/sapmreceiver/go.sum
	deleted:    receiver/signalfxreceiver/go.sum
	deleted:    receiver/simpleprometheusreceiver/go.sum
	deleted:    receiver/statsdreceiver/go.sum
	modified:   receiver/statsdreceiver/testdata/config.yaml
	deleted:    receiver/wavefrontreceiver/go.sum
	deleted:    testbed/go.sum
```

--- a/receiver/statsdreceiver/transport/udp_server.go
+++ b/receiver/statsdreceiver/transport/udp_server.go
@@ -20,7 +20,6 @@ import (
 	"io"
 	"net"
 	"strings"
-	"sync"
 
 	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	"go.opentelemetry.io/collector/consumer"
@@ -31,7 +30,6 @@ import (
 )
 
 type udpServer struct {
-	wg         sync.WaitGroup
 	packetConn net.PacketConn
 	reporter   Reporter
 }
@@ -66,14 +64,9 @@ func (u *udpServer) ListenAndServe(
 	for {
 		n, _, err := u.packetConn.ReadFrom(buf)
 		if n > 0 {
-			u.wg.Add(1)
 			bufCopy := make([]byte, n)
 			copy(bufCopy, buf)
-			// TODO: remove this goroutine spawning
-			go func() {
-				u.handlePacket(parser, nextConsumer, bufCopy)
-				u.wg.Done()
-			}()
+			u.handlePacket(parser, nextConsumer, bufCopy)
 		}
 		if err != nil {
 			u.reporter.OnDebugf("UDP Transport (%s) - ReadFrom error: %v",
@@ -90,9 +83,7 @@ func (u *udpServer) ListenAndServe(
 }
 
 func (u *udpServer) Close() error {
-	err := u.packetConn.Close()
-	u.wg.Wait()
-	return err
+	return u.packetConn.Close()
 }
 
 func (u *udpServer) handlePacket(

